### PR TITLE
Add prestop hook

### PR DESCRIPTION
**Pull request by inteon**
This change is already present on our testing infra ( was added for testing by @jahrlin ).

### CODE REVIEW

File: pkg/prowgen/generators.go [hunk 1]
```diff
@@ -51,6 +51,17 @@ func MakeTest(ctx *ProwContext) *Job {
 					Memory: "4Gi",
 				},
 			},
+			Lifecycle: &Lifecycle{
```
Comment (jahrlin) [inline pkg/prowgen/generators.go:54]: We are mixing pointers and values here, for consistency's sake we should probably only use values

File: pkg/prowgen/types.go [hunk 2]
```diff
@@ -97,6 +99,18 @@ type SecurityContextCapabilities struct {
 	Add []string `yaml:"add"`
 }
 
+type Lifecycle struct {
+	PreStop PreStop `yaml:"preStop"`
+}
+
+type PreStop struct {
```
Comment (jahrlin) [inline pkg/prowgen/types.go:106]: This type should be `LifecycleHandler`, see: https://pkg.go.dev/k8s.io/api/core/v1#Lifecycle

File: pkg/prowgen/generators.go (hunk 1 above)
Comment (munnerz) [reply pkg/prowgen/generators.go:54]: If a non pointer is used then the value can never be unset/nil - some of our jobs don't use a lifecycle hook, and I think those should omit this field together in those cases

**Comment by jahrlin** [comment]
Thanks @inteon!

/lgtm
/approve

**Pull request merged**